logutils/getlogger.go

```diff
@@ -0,0 +1,89 @@
+package logutils
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type loggerConfig struct {
+	devMode bool
+	level   string
+}
+
+// LogConfigOption allows to fine-tune the configuration of the logger.
+type LogConfigOption = func(*loggerConfig)
+
+// LogDevMode tells the logger to work in the development mode.
+func LogDevMode(devMode bool) LogConfigOption {
+	return func(lc *loggerConfig) {
+		lc.devMode = devMode
+	}
+}
+
+// LogLevel sets the desired level of logging.
+func LogLevel(level string) LogConfigOption {
+	return func(lc *loggerConfig) {
+		lc.level = level
+	}
+}
+
+// GetZapLogger returns a logger created according to `log-level` and
+// `log-dev` command-line switches.
+//
+// Note: this method defines two flags, `log-level` and `log-dev`; redefining
+// them in your main app will cause panic.
+func GetZapLogger(options ...LogConfigOption) *zap.Logger {
+	cfg := &loggerConfig{
+		devMode: false,
+		level:   zap.InfoLevel.String(),
+	}
+
+	for _, o := range options {
+		o(cfg)
+	}
+
+	var config zap.Config
+	if cfg.devMode {
+		config = zap.NewDevelopmentConfig()
+	} else {
+		config = zap.NewProductionConfig()
+	}
+
+	config.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
```

Comment:
how's this time format looking like in dev/prod logs?

Reply:
wouldn't we just want a unix timestamp for json logs? not having a strong opinion here

Reply:
from experience, I sometimes follow the logs in their raw stream.  not having to mentally convert from epoch to human-readable is helpful.